Bug: Frontend course-selection.js fails with 'courses.forEach is not a function' after PR #57

## Bug Description

After merging PR #57 (CourseService improvements), the frontend `course-selection.js` file throws a JavaScript error when loading courses.

## Error Details

- **File:** `course-selection.js`
- **Error:** `TypeError: courses.forEach is not a function`
- **Stack trace:**
```
course-selection.js:50 Error loading courses: TypeError: courses.forEach is not a function
    at displayCourses (course-selection.js:62:13)
    at loadCourses (course-selection.js:47:13)
```

## Root Cause

PR #57 changed the `get_all_courses` API endpoint to return a paginated response object:
```json
{
  "items": [...],
  "total": N
}
```

The frontend JavaScript code expects a plain array but is now receiving an object.

## Fix Required

Update `course-selection.js` to access `response.items` instead of using the response directly as an array.

## Related

- Introduced by: PR #57
- Related Issue: #31 (CourseService Improvements)

--- a/app/static/js/admin.js
+++ b/app/static/js/admin.js
@@ -135,8 +135,9 @@ async function fetchCourses() {
         const response = await fetch(API_BASE);
         if (!response.ok) throw new Error('Failed to fetch courses');
         const data = await response.json();
-        // API returns a list directly, not an object with courses property
-        courses = Array.isArray(data) ? data : (data.courses || []);
+        // Handle paginated response format: {items: [...], total: N, ...}
+        // Also support legacy array format for backwards compatibility
+        courses = Array.isArray(data) ? data : (data.items || data.courses || []);
         renderCoursesList();
     } catch (error) {
         showToast('Error loading courses: ' + error.message, 'error');

--- a/app/static/js/course-selection.js
+++ b/app/static/js/course-selection.js
@@ -32,13 +32,17 @@ async function loadCourses() {
         hideNoCourses();
 
         const response = await fetch(API_BASE);
-        
+
         if (!response.ok) {
             throw new Error(`HTTP error! status: ${response.status}`);
         }
 
-        const courses = await response.json();
-        
+        const data = await response.json();
+
+        // Handle paginated response format: {items: [...], total: N, ...}
+        // Also support legacy array format for backwards compatibility
+        const courses = Array.isArray(data) ? data : (data.items || []);
+
         hideLoading();
 
         if (courses.length === 0) {

